### PR TITLE
replica: switched to O_DIRECT

### PR DESCRIPTION
**Pull request by kp6**
based on https://github.com/rancher/sparse-tools/pull/13

### CODE REVIEW

File: replica/diff_disk.go [hunk 1]
```diff
@@ -42,7 +43,7 @@ func (d *diffDisk) WriteAt(buf []byte, offset int64) (int, error) {
 	startSector := offset / d.sectorSize
 	sectors := int64(len(buf)) / d.sectorSize
 
-	c, err := d.files[target].WriteAt(buf, offset)
+	c, err := fio.WriteAt(d.files[target], buf, offset)
```
Comment (ibuildthecloud) [inline replica/diff_disk.go]: Can we avoid the copy and just make the original `[]byte` aligned?  The buffer is originally allocated at https://github.com/rancher/longhorn/blob/master/rpc/wire.go#L68

File: replica/replica.go [hunk 2]
```diff
@@ -223,7 +224,7 @@ func (r *Replica) nextFile(parsePattern *regexp.Regexp, pattern, parent string)
 
 func (r *Replica) openFile(name string, flag int) (*os.File, error) {
 	// TODO: need to turn on O_DIRECT
-	return os.OpenFile(path.Join(r.dir, name), os.O_RDWR|os.O_CREATE|flag, 0666)
+	return os.OpenFile(path.Join(r.dir, name), fio.O_DIRECT|os.O_RDWR|os.O_CREATE|flag, 0666)
```
Comment (ibuildthecloud) [inline replica/replica.go]: I'd prefer to just use `syscall.O_DIRECT`, no reason to abstract it.

**Comment by ibuildthecloud** [comment]
@kp6 Since CI is broken can you just locally run `make ci` and you'll see there are some errors.

File: replica/diff_disk.go (hunk 1 above)
Comment (kp6) [inline replica/diff_disk.go]: Yeah, that's what I was planning to do first but then suspected it was not worth it, tested and saw no difference as I/O masks the copy overhead completely. 
I like that all the alignment stuff is confined to directfio

File: replica/replica.go (hunk 2 above)
Comment (kp6) [inline replica/replica.go]: fixed

File: Dockerfile.dapper [hunk 3]
```diff
@@ -54,12 +54,15 @@ RUN cd /usr/src/tcmu-runner && \
     cp scsi_defs.h /usr/local/include && \
     cp libtcmu_static.a /usr/local/lib/libtcmu.a
 
-# Install ssync
-RUN go get -d github.com/kp6/alphorn/ssync && \
-    cd $GOPATH/src/github.com/kp6/alphorn && \
-    git checkout c7705144deb2d398820fd248c261d2279b27f45d && \
+# Install ssync, etc
```
Comment (ibuildthecloud) [inline Dockerfile.dapper]: I don't think any of these changes are necessary.

File: Dockerfile.dapper [hunk 4]
```diff
@@ -54,10 +54,10 @@ RUN cd /usr/src/tcmu-runner && \
...
+RUN go get -d github.com/rancher/sparse-tools/ssync && \
```
Comment (kp6) [inline Dockerfile.dapper]: what about these?

**Pull request merged**